### PR TITLE
fix(coordinator) - ftx aggregated into correct aggregation interval 

### PR DESCRIPTION
**Pull request by fluentcrafter**
This PR implements issue(s) #2103 

### Checklist

* [ ] I wrote new tests for my new core changes.
* [ ] I have successfully ran tests, style checker and build against my new changes locally.
* [ ] If this change is deployed to any environment (including Devnet), E2E test coverage exists or is included in this
  PR.
* [ ] I have informed the team of any breaking changes if there are any.

---

> [!NOTE]
> **Medium Risk**
> Changes the boundary conditions for which forced transactions are included in aggregation state, which can affect aggregation outputs if mis-specified. Scope is limited and backed by targeted unit tests.
> 
> **Overview**
> Fixes an off-by-one boundary in `AggregationL2StateProviderImpl` so forced transactions are considered **up to and including** the aggregation end block, ensuring the correct FTX is rolled into the aggregation interval.
> 
> Standardizes genesis handling by introducing a shared `GENESIS_ZERO_HASH` and reusing `AnchoredMessage.GENESIS` / `FtxRollingInfo.GENESIS` instead of ad-hoc `ByteArray(32)` defaults, and adds a new `AggregationL2StateProviderImplTest` covering genesis, deployment-before cases, inclusive FTX selection, and block timestamp derivation.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 011d8b0de56eff1716c3a176334822263be1d298. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: coordinator/core/src/test/kotlin/net/consensys/zkevm/ethereum/coordination/aggregation/AggregationL2StateProviderImplTest.kt [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  }
+
+  @Test
+  fun `returns genesis ftx state when no forced transactions exist`() {
+    val blockNumber = 10L
+    ethApiClient.setLatestBlockTag(blockNumber.toULong())
+
+    val state = provider.getAggregationL2State(blockNumber).get()
+
+    assertThat(state.parentAggregationLastFtxNumber).isEqualTo(0uL)
```
Comment (Filter94) [inline coordinator/core/src/test/kotlin/net/consensys/zkevm/ethereum/coordination/aggregation/AggregationL2StateProviderImplTest.kt:76]: Nit: we could refer to `GENESIS` here and in other tests too

**Review by Filter94 — DISMISSED** [review]

**Review by Filter94 — DISMISSED** [review]

**Review by Filter94 — APPROVED** [review]
